baremetal/metal3data_manager.go

```diff
@@ -1440,6 +1440,7 @@ func fetchM3IPClaim(ctx context.Context, cl client.Client, mLog logr.Logger,
 func (m *DataManager) fetchIPClaimsWithLabels(ctx context.Context, pool string) ([]ipamv1.IPClaim, error) {
 	allIPClaims := ipamv1.IPClaimList{}
 	opts := []client.ListOption{
+		&client.ListOptions{Namespace: m.Data.Namespace},
```

Comment:
We need to ~~double~~ triple check that this is OK in all cases! Is there any case where we need to fetch IPClaims from other namespaces than the M3D?

Reply:
True, rather do we have a usecase where we create IPclaims for an M3D in separate namespaces

Reply:
Alright, I have looked through the code a bit more and have some findings.

1. `fetchIPClaimsWithLabels` is only called from `releaseAddressFromM3Pool`
2. I assume the `DataManager` should never release addresses from any other namespaces than the M3D that it is working on at the moment.

Based on that I would say we are quite safe.

Reply:
Hey, thanks for the PR and sorry for the late reply on this one, I was on 2 weeks PTO. 

> 1. `fetchIPClaimsWithLabels` is only called from `releaseAddressFromM3Pool`
> 2. I assume the `DataManager` should never release addresses from any other namespaces than the M3D that it is working on at the moment.

Correct, M3D namespace is tied to IPClaim namespace in all use cases I could think of at the time of implementing the feature.

Reply:
Thanks for confirming @furkatgofurov7 !